modules/sdk-core/src/bitgo/tss/ecdsa/dkls.ts

```diff
@@ -0,0 +1,198 @@
+import { SignatureShareRecord, SignatureShareType } from '../../utils';
+import openpgp from 'openpgp';
+import { DklsComms, DklsTypes } from '@bitgo/sdk-lib-mpc';
+import {
+  MPCv2SignatureShareRound1Input,
+  MPCv2SignatureShareRound1Output,
+  MPCv2SignatureShareRound2Input,
+  MPCv2SignatureShareRound2Output,
+  MPCv2SignatureShareRound3Input,
+  MPCv2PartyFromStringOrNumber,
+} from '@bitgo/public-types';
+import assert from 'assert';
+
+/**
+ Helpers in this take care of all interaction with WP API's
+**/
+
+export async function getSignatureShareRoundOne(
+  round1Message: DklsTypes.DeserializedBroadcastMessage,
+  userGpgKey: openpgp.SerializedKeyPair<string>
+): Promise<SignatureShareRecord> {
+  const serializedMessages = DklsTypes.serializeMessages({
+    broadcastMessages: [round1Message],
+    p2pMessages: [],
+  });
+  const authEncBroadcastMessage = (
+    await DklsComms.encryptAndAuthOutgoingMessages(
+      serializedMessages,
+      [], // Broadcast message so doesn't need to encrypt to BitGo's GPG key
+      [getUserPartyGpgKey(userGpgKey)]
+    )
+  ).broadcastMessages[0];
+  // Share type expected by Wallet Platform's API
+  assert(MPCv2PartyFromStringOrNumber.is(authEncBroadcastMessage.from));
+  const share: MPCv2SignatureShareRound1Input = {
+    type: 'round1Input',
+    data: {
+      msg1: {
+        from: authEncBroadcastMessage.from,
+        message: authEncBroadcastMessage.payload.message,
+        signature: authEncBroadcastMessage.payload.signature,
+      },
+    },
+  };
+  const serializedShare = JSON.stringify(share);
+  return {
+    from: SignatureShareType.USER,
+    to: SignatureShareType.BITGO,
+    share: serializedShare,
+  };
+}
+
+export async function getSignatureShareRoundTwo(
+  userToBitGoMessages2: DklsTypes.DeserializedMessages,
+  userToBitGoMessages3: DklsTypes.DeserializedMessages,
+  userGpgKey: openpgp.SerializedKeyPair<string>,
+  bitgoGpgKey: openpgp.Key
+): Promise<SignatureShareRecord> {
+  const userToBitGoEncryptedMsg2 = await DklsComms.encryptAndAuthOutgoingMessages(
+    DklsTypes.serializeMessages(userToBitGoMessages2),
+    [getBitGoPartyGpgKey(bitgoGpgKey)],
+    [getUserPartyGpgKey(userGpgKey)]
+  );
+
+  const userToBitGoEncryptedMsg3 = await DklsComms.encryptAndAuthOutgoingMessages(
+    DklsTypes.serializeMessages(userToBitGoMessages3),
+    [getBitGoPartyGpgKey(bitgoGpgKey)],
+    [getUserPartyGpgKey(userGpgKey)]
+  );
+  assert(MPCv2PartyFromStringOrNumber.is(userToBitGoEncryptedMsg2.p2pMessages[0].from));
+  assert(MPCv2PartyFromStringOrNumber.is(userToBitGoEncryptedMsg2.p2pMessages[0].to));
+  assert(MPCv2PartyFromStringOrNumber.is(userToBitGoEncryptedMsg3.p2pMessages[0].from));
+  assert(MPCv2PartyFromStringOrNumber.is(userToBitGoEncryptedMsg3.p2pMessages[0].to));
```

Comment:
wont all these throw a `cannot access foo from undefined `if the array is empty?

Reply:
maybe assert that exists and use a codec to check the entire msgs?

Reply:
added a check to see the length before. I have a TODO to decode the API messages using codec, i tried that but for some reason the decode gives an error even though the response matches the codec. I'm gonna do this as a follow-up.